### PR TITLE
OSDOCS-2978: Upstream Kubernetes Overview

### PR DESCRIPTION
**Pull request by xenolinux**
Adds an overview of upstream Kubernetes

Version(s): 4.6+

Issue:  https://issues.redhat.com/browse/OSDOCS-2978 

Link to docs preview: http://file.rdu.redhat.com/sdudhgao/OSDOCS-2978/getting_started/kubernetes-overview.html

Additional information:

### CODE REVIEW

File: getting_started/kubernetes-overview.adoc [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+To solve this problem, you can use containerization technologies that segregate applications in a containerized environment. Similar to a VM, a container has its own filesystem, vCPU, memory, process space, dependencies, and more. Containers are decoupled from the underlying infrastructure, and are portable across clouds and OS distributions. Containers are inherently much lighter than a fully-featured OS, and are light weight isolated processes that run on the operating system kernel. VMs are slower to boot, and are an abstraction of physical hardware. VMs run on a single machine with the help of Hypervisor.
+
+You can perform the following actions by using Kubernetes:
+
+* Sharing resources.
+* Orchestrating containers across multiple hosts.
+* Installing new hardware configurations.
+* Running health checks and self-healing applications.
+* Scaling containerized applications.
+
```
Comment (ahardin-rh) [inline getting_started/kubernetes-overview.adoc:25]: @xenolinux I think that Kubernetes components can be its own conceptual module, which can help break up the overview a bit and allow for potential content resuse

File: getting_started/kubernetes-overview.adoc [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+.Kubernetes cluster overview
+image::247_OpenShift_Kubernetes_Overview-1.png[]
+
+A custom resource is an extension of the Kubernetes API. You can customize Kubernetes clusters by using custom resources. Operators are software extensions which manage applications and their components with the help of custom resources. Kubernetes uses a declarative model when you want a fixed desired result while dealing with cluster resources. By using Operators, Kubernetes defines its states in a declarative way. You can modify the Kubernetes cluster resources by using imperative commands.
```
Comment (ahardin-rh) [inline getting_started/kubernetes-overview.adoc:73]: Same here. I think Kubernetes resources can be its own conceptual module.

File: getting_started/kubernetes-overview.adoc [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+.Kubernetes Resources
+[cols="1,2",options="header"]
+|===
+|Resource |Purpose
+
+|Service
+|Kubernetes uses services to expose a running application on a set of pods.
+
+|ReplicaSets
+|Kubernetes uses ReplicaSets to maintain the constant pod number.
```
Comment (ahardin-rh) [inline getting_started/kubernetes-overview.adoc:85]: Kubernetes uses the `ReplicaSets` resource

File: getting_started/kubernetes-overview.adoc [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+
+|Deployment
+|A resource object that maintains the life cycle of an application.
+|===
+
+Kubernetes is a core component of an {product-title}. You can use {product-title} for developing and running containerized applications. With its foundation in Kubernetes, the {product-title} incorporates the same technology that serves as the engine for massive telecommunications, streaming video, gaming, banking, and other applications. You can extend your containerized applications beyond a single cloud to on-premise and multi-cloud environments by using the {product-title}.
+
+.Architecture of Kubernetes
+image::247_OpenShift_Kubernetes_Overview-2.png[]
+
```
Comment (ahardin-rh) [inline getting_started/kubernetes-overview.adoc:95]: This "before getting started chunk of content" could be its own conceptual module called Conceptual guidelines or something similar

File: getting_started/kubernetes-overview.adoc [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+|Stores the data even after the device is shut  off. Kubernetes uses persistent volumes to store the application data.
+
+|`container-registry`
+|Stores and accesses the container images.
+
+|Pod
+|The pod is the smallest logical unit in Kubernetes. A pod contains one or more containers to run in a worker node.
+|===
+
+A cluster is a single computational unit consisting of multiple nodes in a cloud environment. A Kubernetes cluster includes a control plane and worker nodes. You can run Kubernetes containers across various machines and environments. The control plane node controls and maintains the state of a cluster. You can run the Kubernetes application by using worker nodes. You can use the Kubernetes namespace to differentiate cluster resources in a cluster. Namespace scoping is applicable for resource objects, such as deployment, service, and pods. You can not use namespace for cluster-wide resource objects such as `storageclass`, nodes, and persistent volumes.
```
Comment (abhatt-rh) [inline getting_started/kubernetes-overview.adoc:28]: Per [OpenShift doc guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-references), When referring to API object terms in general usage, use lowercase and separate multi-word API objects.
```suggestion
A cluster is a single computational unit consisting of multiple nodes in a cloud environment. A Kubernetes cluster includes a control plane and worker nodes. You can run Kubernetes containers across various machines and environments. The control plane node controls and maintains the state of a cluster. You can run the Kubernetes application by using worker nodes. You can use the Kubernetes namespace to differentiate cluster resources in a cluster. Namespace scoping is applicable for resource objects, such as deployment, service, and pods. You cannot use namespace for cluster-wide resource objects such as storage class, nodes, and persistent volumes.
```

File: getting_started/kubernetes-overview.adoc (hunk 2 above)
Comment (xenolinux) [reply getting_started/kubernetes-overview.adoc]: Done

File: getting_started/kubernetes-overview.adoc (hunk 1 above)
Comment (xenolinux) [reply getting_started/kubernetes-overview.adoc:25]: Done

File: getting_started/kubernetes-overview.adoc [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+* Create special kinds of assets. For example, services are represented by a set of pods and a policy that defines how they are accessed. This policy allows containers to connect to the services that they need even if they do not have the specific IP addresses for the services. Replication controllers are another special asset that indicates how many pod replicas are required to run at a time. You can use this capability to automatically scale your application to adapt to its current demand.
+
+The API to the {product-title} cluster is 100% Kubernetes. Nothing changes between a container running on any other Kubernetes and running on the {product-title}. No changes to the application.
+The {product-title} brings added-value features to provide enterprise-ready enhancements to Kubernetes. The {product-title} CLI tool  (`oc`) is compatible with `kubectl`. While the Kubernetes API is 100% accessible within {product-title}, the `kubectl` command-line lacks many features that could make it more user-friendly. The {product-title} Container Platform offers a set of features and command-line tool like OC.
```
[the remark this answers is not in the document]
Comment (xenolinux) [reply getting_started/kubernetes-overview.adoc]: Done

File: getting_started/kubernetes-overview.adoc (hunk 3 above)
Comment (xenolinux) [reply getting_started/kubernetes-overview.adoc]: Done

**Review by abhatt-rh — COMMENTED** [review]
Hi Sarvesha, nice work on providing an overview for K8s
I agree with @ahardin-rh's comments. Aside from those, I left a couple of more suggestions and comments for your consideration. Thanks!

**Comment by r2munz** [comment]
Hi Sarvesha, excellent job in putting all this together. Looks good to me, approved.

File: modules/kubernetes-components.adoc [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+|Stores cluster data.
+
+|`kube-apiserver`
+|Validates and configures data for the API objects.
+
+|`kubelet`
+|Runs on nodes and reads the container manifests. Ensures that the defined containers have started and are running.
+
+|`kubectl`
+|Allows you to define how you want to run workloads. Use the `kubectl` command to interact with the `kube-api-server`.
```
Comment (abhatt-rh) [inline modules/kubernetes-components.adoc:33]:
```suggestion
|Allows you to define how you want to run workloads. Use the `kubectl` command to interact with the `kube-apiserver`.
```

File: modules/kubernetes-conceptual-guidelines.adoc [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+* Create special kinds of assets. For example, services are represented by a set of pods and a policy that defines how they are accessed. This policy allows containers to connect to the services that they need even if they do not have the specific IP addresses for the services. Replication controllers are another special asset that indicates how many pod replicas are required to run at a time. You can use this capability to automatically scale your application to adapt to its current demand.
+
+The API to {product-title} cluster is 100% Kubernetes. Nothing changes between a container running on any other Kubernetes and running on {product-title}. No changes to the application.
+{product-title} brings added-value features to provide enterprise-ready enhancements to Kubernetes. {product-title} CLI tool  (`oc`) is compatible with `kubectl`. While the Kubernetes API is 100% accessible within {product-title}, the `kubectl` command-line lacks many features that could make it more user-friendly. {product-title} offers a set of features and command-line tool like OC.
```
Comment (ahardin-rh) [inline modules/kubernetes-conceptual-guidelines.adoc:17]: OC should be `oc`

**Pull request merged**